website/server/controllers/api-v3/groups.js

```diff
@@ -344,7 +345,14 @@ api.getGroups = {
     }
 
     if (req.query.member) {
-      filters._id = { $in: user.guilds };
+      // Includes 'guilds' nor 'publicGuilds' type.
+      if (req.query.type === 'publicGuilds' || req.query.type === 'guilds') {
```

Comment:
this could be simplified by doing something like this outside of the if block

```
if (!filters._id) {
    filters._id = {};
}

filters._id['$ne'] = TAVERN_ID;
```

so that multiple cases depending on whether we're fetching guilds or publicGuilds aren't necessary

Reply:
Ah I was looking for documentation on how to do that one properly with Mongoose, since I hadn't touched it before.

What exactly did you mean with:
> so that multiple cases depending on whether we're fetching guilds or publicGuilds aren't necessary

The documentation for type parameter says:
> Possible values are party, guilds, privateGuilds, publicGuilds, tavern.

Did you mean with this that the tavern should never be fetched? Because I was under the impression that it could be fetched with the `tavern` type parameter. Unless I am misunderstanding the code, it will never be able to fetch the Tavern if `filters._id['$ne'] = TAVERN_ID;` would always be used to filter it out, even with the `tavern` type parameter.

Reply:
ah good point... I think the problem might go away when the check is moved inside Group.getGroups since you'll have to filter for TAVERN_ID in the two switch cases about `guilds` and `publicGuilds` so the tavern one should not be touched at all